### PR TITLE
Etv 3558 documentation

### PR DESCRIPTION
**Pull request by alsofronie**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,58 +43,615 @@ Use a `<script>` tag to reference the package. Using this method will register a
...
 ```
 
-Although not recommended, if you need to display it in multiple div's, identified by a common class, you can use:
+## Initializing the library
+
+The library needs some configuration in order to work and communicate properly with our backend services. We do that by calling a function named `initialize`. You will want to call this function as early as possible in your code, making sure it is the first one called:
+
+```
+automat.initialize({
+  etvasURL: "https://appname.etvasapps.com",
```
Comment (ohnu) [inline README.md:92]: I do not get what url is this ? it seems like an app url but which app ?

Comment (ohnu) [reply README.md:92]: ohh I've read that is the customer portal url .. maybe change the domain to helloetvas then ?

File: README.md [hunk 2]
```diff
@@ -39,58 +43,615 @@ Use a `<script>` tag to reference the package. Using this method will register a
...
+
+Only the `etvasURL` and the `organizationId` are required. The other ones have either a default value or they are needed just in some cases. Aside the `locale` entry, everything else is provided by Etvas Support Team or you can find the values in [Etvas Partners Portal](https://partners.helloetvas.com)
+
+#### `etvasURL`
+
+This is the URL you can find in your account on [Etvas Partner's Studio](https://partners.helloetvas.com). We call this URL the Customer Portal, the portal tailored and branded by you to show it to your customers. It is required in order to preserve the brand colors and graphics you provided, so the integration in your application is as natural as possible.
+
+#### `etvasApiKey`
+
+This is a public key we provide (and you can find it in your account in [Etvas Partner's Studio](https://partners.helloetvas.com)) which is used not only to uniquely identify your app (the public part) but also to watch, analyze and shape the incoming traffic to automatically protect against a wide range of network attacks. Please note the entire process is anonymous and no personal data is stored by these operations.
```
Comment (ohnu) [inline README.md:107]: is this the Partner API key or the graphql public api key ?

File: README.md [hunk 3]
```diff
@@ -39,58 +43,615 @@ Use a `<script>` tag to reference the package. Using this method will register a
...
 
-Please note, if you are using jQuery, please use the following syntax, in order to have the native JS DOM element and not a jQuery wrapper:
+where `callback` is a function that will run when successfully connected.
+
+The function accepts a callback which will be called without any arguments but only on successful login:
 
 ```
-const $element = $('#etvas-placeholder')
-etvasAutomat.showProductCard($element.first()[0], options)
+automat.connect(function () {
```
Comment (ohnu) [inline README.md:142]: how do I know if there was an error on connect ?

Comment (alsofronie) [reply README.md:142]: This is out of our hands, we can only open the window (it's managed by Cognito). We can only detect an auth, in which case the callback is called. No errors whatsoever.

File: README.md (hunk 2 above)
Comment (alsofronie) [reply README.md:107]: Its the graphql one I think (`da2-xxx`).

Comment (ohnu) [reply README.md:107]: in that case I don't believe we expose it in the Partner's Studio ... 
that one is a global API Key ... not even one per partner :|

**Review by ohnu — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (alsofronie) [reply README.md:107]: Yes. I think this is the API key that is used by our customer portal(s) to make unauthenticated calls. Unfortunately the `getBrandingBySlug` GraphQL query falls into that category.

Being this case, I really don't want to give them this API key to them. We should discuss options.

Comment (ohnu) [reply README.md:107]: One option would be to: Use it internally in etvas-automat and not document that you can override it.

Comment (alsofronie) [reply README.md]: Done

**Review by ohnu — APPROVED** [review]

**Pull request merged**